project-metrics/metrics_service/apis/travis.py

```diff
@@ -26,22 +28,36 @@ def __init__(self, status_code: int, err_msg: Text):
 class TravisApi(agithub_base.API):
   """Travis API interface."""
 
-  def __init__(self, *args, **kwargs):
+  # Default/maximum page size from the API.
+  TRAVIS_PAGE_SIZE = 25
 
+  def __init__(self, *args, **kwargs):
     extra_headers = {
         'User-Agent': 'AMPProjectMetrics/1.0.0',
-        'Accept': 'application/vnd.travis-ci.2.1+json',
-        'Content-Type': 'application/json'
+        'Content-Type': 'application/json',
     }
+
     props = agithub_base.ConnectionProperties(
         api_url='api.travis-ci.org',
         secure_http=True,
         extra_headers=extra_headers)
     self.setClient(agithub_base.Client(*args, **kwargs))
     self.setConnectionProperties(props)
 
-    self._token = self._get_travis_token(env.get('GITHUB_API_ACCESS_TOKEN'))
+    try:
+      self.api_token = env.get('TRAVIS_API_ACCESS_TOKEN')
+    except KeyError:
+      logging.warn('No Travis API key found; exchanging GitHub token')
+      self.api_token = self._get_travis_token(
+          env.get('GITHUB_API_ACCESS_TOKEN'))
+
     extra_headers['Authorization'] = 'token %s' % self._token
+    extra_headers['Travis-API-Version'] = '3'
```

Comment:
Can add this where you define the `extra_headers`

Reply:
Nope, the `auth` API is V2, so this gets added after the credential exchange. That said, it seems like once you exchange a GitHub token for a Travis token, the Travis token doesn't change. So, it may even be worth removing all this token exchange logic, requiring a Travis token in the YAML, and providing the CLI command to generate a Travis token from a GitHub token. Thoughts?

Reply:
Hmm... Yeah? I think that makes sense

In your summary doc write something about how the Travis token was auto-generated and doesn't seem to change, but if it does "here's the code I had before to auto-generate it on the fly"

Reply:
Done